app/src/lib/git/lfs.ts

```diff
@@ -34,3 +35,76 @@ export async function isUsingLFS(repository: Repository): Promise<boolean> {
   })
   return result.stdout.length > 0
 }
+
+/**
+ * Check if a provided file path is being tracked by Git LFS
+ *
+ * This uses the Git plumbing to read the .gitattributes file
+ * for any LFS-related rules that are set for the file
+ *
+ * @param repository repository with
+ * @param path relative file path in the repository
+ */
+export async function isTrackedByLFS(
+  repository: Repository,
+  path: string
+): Promise<boolean> {
+  const { stdout } = await git(
+    ['check-attr', 'filter', path],
+    repository.path,
+    'checkAttrForLFS'
+  )
+
+  // "git check-attr -a" will output every filter it can find in .gitattributes
+  // and it looks like this:
+  //
+  // README.md: diff: lfs
+  // README.md: merge: lfs
+  // README.md: text: unset
+  // README.md: filter: lfs
+  //
+  // To verify git-lfs this test will just focus on that last row, "filter",
+  // and the value associated with it. If nothing is found in .gitattributes
+  // the output will look like this
+  //
+  // README.md: filter: unspecified
+
+  const lfsFilterRegex = /(.*): filter: lfs/
+
+  const match = lfsFilterRegex.exec(stdout)
```

Comment:
It looks like you're calling `git check-attr filter` instead of `git check-attr -a`, so (if my reading of the surrounding code is correct) I think that Git will only report about the `filter` attribute. Thus, I think this regex match can removed in favor of simply comparing to and dropping the `: filter: lfs` suffix.

Reply:
@ttaylorr I'm using the capture as well as a sanity check that it's the relative file path I passed in as an argument, but if I decide that I don't need to do that I can simplify the regex 👍